protocol/messenger_handler.go

```diff
@@ -410,10 +410,89 @@ func (m *Messenger) handleCommandMessage(state *ReceivedMessageState, message *c
 	return nil
 }
 
+func (m *Messenger) syncReceivedContactRequest(contact *Contact, state *ReceivedMessageState, chat *Chat) error {
+	contactRequestID, err := m.persistence.LatestPendingContactRequestIDForContact(contact.ID)
+	if err != nil {
+		return err
+	}
+
+	if contactRequestID != "" {
+		return nil
+	}
+
+	if chat == nil {
+		return fmt.Errorf("no chat restored during the contact synchronisation, contact.ID = %s", contact.ID)
+	}
+
+	clock, timestamp := chat.NextClockAndTimestamp(m.transport)
+	contactRequest, err := m.generateContactRequest(clock, timestamp, contact, "Please add me to your contacts", false)
```

Comment:
I found 4 usages of this magical sentence `Please add me to your contacts` in production code. Since this function is also duplicating this string, it would be a good opportunity to extract them all into a constant.

Reply:
Absolutely @caybro. Ideally, I think the text used to generate the contact request message should always come from the client, so it's the client's responsibility to translate and respect the user's preferences. To me, the endpoint `SendContactRequest` should even validate the presence of the message, and if it's not there not default to anything and err, since English doesn't make any sense for a lot of users.

But of course, not something I'm suggesting for this PR.

Reply:
I hope in future we should restore CR message with valid ID and Message, and default contact request text would be used only for backward compatibility. In desktop code we can provide transition  with QT_TR_NOOP for external strings

Reply:
fixed